src/core/api_client/tests/eventHandling.test.ts

```
@@ -0,0 +1,196 @@
+
+      // Make multiple concurrent requests
+      const promises = [
+        context.apiClient.get('/test', undefined, false),
+        context.apiClient.post('/test', { data: 'test' }, false),
+        context.apiClient.get('/test', undefined, false),
+      ];
+
+      // Should have pool updates with multiple pending requests
+      const pendingPool = poolUpdates.find((pool) => pool.size > 1);
+      expect(pendingPool).toBeDefined();
+      expect(pendingPool!.size).toBeGreaterThan(1);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)